eval/mt_bench_pipeline.py

```diff
@@ -0,0 +1,129 @@
+# mt_bench_pipeline.py
+import inspect
+from kfp import dsl, compiler
+from kfp import kubernetes
+
+IMAGE = "quay.io/sallyom/instructlab-ocp:eval"  # UPDATE if needed
+PIPELINE_ROOT = "s3://somaltest"  # UPDATE if needed
+
+@dsl.container_component
+def run_mt_bench_script(
+    model_s3_path: str,
+    eval_script_s3_path: str,
+    judge_model_name: str,
+    server_url: str,
+    output_dir: str,
+    output_s3_path: str,
+    access_key: str,
+    secret_key: str,
+    session_token: str = "",
+    openai_api_key: str = "",
+    max_workers: str = "auto",
+    merge_system_user_message: bool = False,
+):
+    """
+    Runs mt_bench script inside the pipeline container with AWS credentials.
+    """
+
+    script = inspect.cleandoc(
+        f"""
+        export AWS_ACCESS_KEY_ID={access_key}
+        export AWS_SECRET_ACCESS_KEY={secret_key}
+        export AWS_SESSION_TOKEN={session_token}
+        export OPENAI_API_KEY={openai_api_key}
+        export PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True
+
+        python3.11 -m pip install --upgrade pip
+        python3.11 -m pip install boto3
+
+        # Python script to download mt_bench script from S3
+        python3.11 -c "
+import boto3
+import os
+
+aws_access_key = os.getenv('AWS_ACCESS_KEY_ID')
+aws_secret_key = os.getenv('AWS_SECRET_ACCESS_KEY')
+aws_region = os.getenv('AWS_REGION', 'us-east-1')
+s3_endpoint_url = os.getenv('S3_ENDPOINT_URL', 'https://s3.amazonaws.com')
+
+s3_client = boto3.client(
+    's3',
+    aws_access_key_id=aws_access_key,
+    aws_secret_access_key=aws_secret_key,
+    region_name=aws_region,
+    endpoint_url=s3_endpoint_url,
+)
+
+eval_script_s3_path = '{eval_script_s3_path}'
+local_path = '/instructlab/mt_bench.py'
+print(f'Attempting to download mt_bench script from {{eval_script_s3_path}} to {{local_path}}')
+
+bucket_name = eval_script_s3_path.split('/')[2]
+key = '/'.join(eval_script_s3_path.split('/')[3:])
+
+try:
+    s3_client.download_file(bucket_name, key, local_path)
+    print(f'Downloaded {{key}} to {{local_path}}')
+except Exception as e:
+    print(f'Error downloading mt_bench script from S3: {{e}}')
+    raise
```

Comment:
This is the 3rd time you're downloading from S3, using yet another code snippet - 3rd way how to download a thing... This time in a string. This is definitely a danger.

Reply:
> WIP: Don't Review!! Don't try to run (probably won't run anyways) :)

We don't need S3 at all, I replaced with importer and Artifacts